notebooks/flusight_from_idata.qmd

```diff
@@ -0,0 +1,233 @@
+
+### `output_type_id`
+
+> This column holds any needed information about what subclass of the `output type` the row contains. For `quantile`, it contains the **level of the quantile** (e.g. `0.01` for the 1st percentile, `0.5` for the median / 50th percentile, etc). For `pmf`, contains the **name of the category/event whose probability is being given** (e.g. `'stable'` or `'large decrease'`).
+
+### `value`
+
+> This column contains the actual forecast value. For a `quantile` forecast of a hospitalizations `target`, this is the number of forecast hospitalizations for the given `location` and `horizon` at the given quantile level (specified in `output_type_id`)
+
+## Example Forecast From Arviz InferenceData
+
+Suppose that a model has produced a 28 day forecast of influenza hospitalizations for Texas[^individual_forecast] using a spline regression model (fit on daily resolution NHSN hospital admissions data from `2022/08/08` to `2023/12/08`) in `numpyro`.
+
+[^individual_forecast]: While it is possible to create an `idata` object containing posterior samples for all jurisdictions, different `idata` objects allows for modularity with respect to jurisdiction-specific anomaly detection and handling. Thus, each `idata` object corresponds to forecasts for a single jurisdiction, rather than all jurisdictions.
+
+Associated with the model is an [mcmc instance](https://num.pyro.ai/en/stable/mcmc.html); the forecast is captured within the posterior [predictive](https://num.pyro.ai/en/stable/utilities.html#predictive) samples.
+
+The mcmc instance and posterior predictive samples have be passed to `arviz.from_numpyro()` to produce an `InferenceData` object.
+
+Let us examine this object.
+
+```{python}
+# | label: examine-inference-data-object
+#| echo: true
+import xarray as xr
+
+
+xr.set_options(display_expand_data=False, display_expand_attrs=False)
+
+# load example forecast(s)
+idata = forecasttools.nhsn_flu_forecast
+idata
+```
+
+The `idata` object has several groups associated with it.
+
+Let us examine the observed data group and then plot out the observations.
+
+```{python}
+# examine the observed data (dimensions, coordinates, data variables)
+idata.observed_data
+```
+
+```{python}
+# | label: examine-observed-data
+import matplotlib.pyplot as plt
+
+
+# examine the observations
+fig, ax = plt.subplots(1, 1, figsize=(8,6))
+data = idata.observed_data["obs"].to_numpy()
+ax.plot(
+    list(range(len(data))),
+    data,
+    color="black",
+    linewidth=1.0,
+    markersize=3.0,
+    marker="o"
+)
+ax.set_ylabel("Hospital Admissions")
+ax.set_xlabel("Time (Days)")
+plt.show()
+```
+
+Observation 1: By default, dates are not associated with any group. As such, either a list of dates corresponding to fitting and forecast periods must be generated from the initial and final fitting dates and then integrated into a dataframe OR dates must be added to the `idata` object itself.
```

Comment:
I would like to think about including dates in the `idata` objects we export.

Reply:
Check out [here](https://github.com/CDCgov/forecasttools-py/pull/10/files#diff-933e4a0af2fb128add33510235f3688c7b5ce7545f51b4b4cb335b1da81a05c7). I tried something different and this seems to work as a way of associating dates with an `idata` object.